File: fixtures/javascript-upb03-dependency/math/sum.js
/** @param {bigint} value @returns {bigint} */
function normalize(value) {
  return BigInt.asIntN(64, value);
}

/** @param {bigint} left @param {bigint} right @returns {bigint} */
export function Sum(left, right) {
  return normalize(BigInt.asIntN(64, left + right));
}

File: fixtures/lua-upb02-modules/math/sum.lua
local Seme = assert(_G.Seme, "Seme adapters are required")

---@seme-id 80222222222222222222222222222201
---@param value seme.i64
---@return seme.i64
local function normalize(value)
  return Seme.add(value, Seme.i64_literal("0"))
end

---@seme-id 80222222222222222222222222222202
---@param left seme.i64
---@param right seme.i64
---@return seme.i64
local function Sum(left, right)
  return Seme.add(left, right)
end

return { Sum = Sum }
